src/main/java/org/jenkinsci/plugins/workflow/graph/BlockEndNode.java

```diff
@@ -71,4 +77,24 @@ public BlockEndNode(FlowExecution exec, String id, START start, List<FlowNode> p
         return start;
     }
 
+    @CheckForNull
+    @Override
+    public Result getStatus() {
+        ErrorAction errorAction = getError();
+        if (errorAction != null) {
+            if (errorAction.getError() instanceof FlowInterruptedException) {
+                return Result.ABORTED;
+            } else {
+                return Result.FAILURE;
+            }
+        } else {
+            FlowNodeStatusAction statusAction = getPersistentAction(FlowNodeStatusAction.class);
+            if (statusAction != null) {
+                return statusAction.getResult();
+            }
+        }
+        // If there's no error on the end node or status action on the end node, return success.
+        return Result.SUCCESS;
```

Comment:
May be better to return null (or GenericStatus.IN_PROGRESS if we swap the type) if node is not completed.

Reply:
Since this is the end node, I'm working under the assumption that it's complete if it exists at all.

Reply:
That's not true -- remember there's a window of time when a FlowNode is being configured before it's completed, and we also have nested block structures doing various things.

Reply:
Okiedokie, then I'll change it to return null (for now, eventually `GenericStatus.IN_PROGRESS`) if `isActive()` is true and `Result.SUCCESS` (until `GenericStatus` blah blah etc) if not.